packages/core/src/payment/strategies/stripev3/stripev3-payment-strategy.ts

```diff
@@ -516,13 +522,21 @@ export default class StripeV3PaymentStrategy implements PaymentStrategy {
             try {
                 return await this._store.dispatch(this._paymentActionCreator.submitPayment(paymentPayload));
             } catch (error) {
-                throw this._handleEmptyPaymentIntentError(error, result.error);
+                throw this._handleEmptyPaymentIntentError(error, result?.error);
             }
         }
 
         throw error;
     }
 
+    private _getPaymentToken(paymentIntent: PaymentIntent | undefined, clientSecret: string, catchedConfirmError: boolean): string {
+        if (!paymentIntent || catchedConfirmError) {
+            return clientSecret;
+        }
+
+        return paymentIntent.id || '' ;
```

Comment:
Again do we need to return an empty string as a token? Is that a valid value?

Reply:
That's right. An empty string is not a valid value.